layers/core_checks/cc_external_object.cpp

```diff
@@ -556,3 +556,150 @@ bool CoreChecks::PreCallValidateExportMetalObjectsEXT(VkDevice device, VkExportM
+    if (import_memory_metal_info == nullptr) {
+        return skip;
+    }
+
+    // Ensure user is importing correct type before checking any other VUID since we will assume correct type
+    if ((import_memory_metal_info->handleType != VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLBUFFER_BIT_EXT) &&
+        (import_memory_metal_info->handleType != VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT) &&
+        (import_memory_metal_info->handleType != VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLHEAP_BIT_EXT)) {
+        skip |= LogError("VUID-VkImportMemoryMetalHandleInfoEXT-handleType-10410", device,
+                         allocate_info_loc.dot(Struct::VkImportMemoryMetalHandleInfoEXT, Field::handleType), "current value is %s",
+                         string_VkExternalMemoryHandleTypeFlagBits(import_memory_metal_info->handleType));
+        return skip;
+    }
+
+    // Only images require to be created as a dedicated allocation. This allows us to check if the format
+    // used by the image allows for importing such images from external memory. We cannot do that with
+    // buffers since we lack the usage, so it's not possible to know if the device allows import
+    // operations on buffers.
+    if (import_memory_metal_info->handleType == VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT) {
+        if (allocate_info.allocationSize != 0) {
+            skip |= LogError("VUID-VkMemoryAllocateInfo-pNext-10397", device,
+                             allocate_info_loc.dot(Field::allocationSize), "VkMemoryAllocateInfo::allocationSize is %" PRId64,
+                             allocate_info.allocationSize);
+        }
+
+        if (dedicated_allocation_info == nullptr) {
+            skip |= LogError("VUID-VkMemoryAllocateInfo-pNext-10395", device,
+                             allocate_info_loc.dot(Field::pNext),
+                             "VK_EXTERNAL_MEMORY_HANDLE_TYPE_MTLTEXTURE_BIT_EXT requires textures to be imported as a dedicated"
+                             "allocation.");
+            // Early out since the image comes from VkMemoryDedicatedAllocateInfoKHR and there's none.
+            return skip;
+        }
+
+        // Unsure if there should be a VUID that enforces image to not be NULL when importing a MTLTEXTURE type
+        if (dedicated_allocation_info->image == VK_NULL_HANDLE) {
+            skip |= LogError("VUID-VkMemoryAllocateInfo-pNext-10395", device,
+                             allocate_info_loc.dot(Struct::VkMemoryDedicatedAllocateInfo, Field::image),
+                             "must be a valid image handle.");
+            // Early out since there's no image in VkMemoryDedicatedAllocateInfoKHR.
+            return skip;
+        }
+
+        // We can only validate images since we lack information to do the buffer call for the properties
+        auto image_state_ptr = Get<vvl::Image>(dedicated_allocation_info->image);
+        VkFormat image_format = image_state_ptr->safe_create_info.format;
```

Comment:
```suggestion
        VkFormat image_format = image_state_ptr->create_info.format;
``` 

you dont need to use the safe version here, they reference the same thing